### PR TITLE
Reorganizes the js modules

### PR DESCRIPTION
**Pull request by joker-777**
* moves each class into its own files and let alephbet.js import them all
* also splits the jest tests for them
* rename AlephbetAdapter to TrackingAdapter to give it a more context
  specific name
* remove lamed adapter and its tests since it is basically the same as the gimel adapter
* removes all build files from dist except the two .min files
* added a  `prepare` script which creates them when creating a package
* the `commonjs` files are now build into its own folder
* commonjs and esm build files are now git ignored and because of this I
  added .npmignore which doesn't ignore them when they get packaged
* remove `vender/lodash.custom.min.js` because we don't use it anymore

### CODE REVIEW

File: src/adapters/gimel_adapter.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import utils from "../utils"
+
+// # Adapter for using the gimel backend. See https://github.com/Alephbet/gimel
+// # The main difference is the user_uuid generation (for backwards compatibility)
+// # params:
+// # - url: Gimel track URL to post events to
+// # - namepsace: namespace for Gimel (allows setting different environments etc)
+// # - storage (optional) - storage adapter for the queue
+class GimelAdapter extends TrackingAdapter {
+  static queue_name = "_lamed_queue"
```
Comment (gingerlime) [inline src/adapters/gimel_adapter.js:11]: this should be `_gimel_queue`? was this bug on the original code?

**Review by gingerlime — CHANGES_REQUESTED** [review]
I think it's worth keeping the lamed adapter tests, even though they are effectively the same... Unless we can explicitly test the parent adapter, and then only test the differences with the Gimel adapter?

I also think `AlephbetAdapter` is a better name, because there are many different tracking adapters, but those are specific to the alephbet "organization" (alephbet, gimel, lamed etc). TrackingAdapter can be anything really, including Google etc.

File: src/adapters/gimel_adapter.js (hunk 1 above)
Comment (joker-777) [reply src/adapters/gimel_adapter.js:11]: Yes, it was like this already in the original code. I will change it.

**Comment by joker-777** [comment]
Yes, I think we should write tests specifically for the `TrackingAdapter` and only test difference for the `GemelAdapter`. About the name, eventually any of those adapters you will pass as a `tracking_adapter` attribute to the experiment, right? That's why I thought using `TrackingAdapter` as the main adapter makes kind of sense.

**Comment by gingerlime** [comment]
Yes, these are all tracking adapters, and that's why I don't think the name makes sense. It's too generic.

File: src/adapters/gimel_adapter.js (hunk 1 above)
Comment (joker-777) [reply src/adapters/gimel_adapter.js]: Updated it.

**Comment by joker-777** [comment]
@gingerlime I renamed TrackingAdapter to AlephbetAdapter and also updated the tests as we discussed.

**Comment by gingerlime** [comment]
If I test the same html as in #43 with this branch, then I get some weird data in localStorage and sent to Gimel/Lamed (It tries to send `http://example.com/?0=undefined&1=undefined&2=undefined&3=undefined&4=undefined`)

CleanShot 2021-01-08 at 08 32 57

I'm not sure if those problems are in master or on this branch though

**Comment by joker-777** [comment]
@gingerlime Update the PR with the current master

**Review by gingerlime — APPROVED** [review]
👍 thank you!

**Comment by gingerlime** [comment]
can you merge and release a new version?

**Comment by joker-777** [comment]
@gingerlime I somehow can't publish it ;( 
Screenshot_20210112_085043

**Comment by joker-777** [comment]
I think I have to first login into the npm account to publish it.

**Pull request merged**